harness/determined/pytorch/_pytorch_context.py

```diff
@@ -368,18 +372,28 @@ def wrap_scaler(self, scaler: Any) -> Any:
             The scaler. It may be wrapped to add additional functionality for use in Determined.
         """
 
-        check.true(HAVE_AMP, "Failed to import torch.cuda.amp. PyTorch >= 1.6 required.")
+        if not HAVE_AMP:
+            raise det.errors.InvalidExperimentException(
+                "Failed to import torch.cuda.amp. PyTorch >= 1.6 required.",
+            )
 
-        check.false(self._use_apex, "Do not mix APEX with PyTorch AMP.")
+        if self._use_apex:
+            raise det.errors.InvalidExperimentException("Do not mix APEX with PyTorch AMP.")
 
-        check.is_none(self._scaler, "Please only call wrap_scaler or use_amp once.")
+        if self._scaler is not None:
+            raise det.errors.InvalidExperimentException(
+                "Please only call wrap_scaler or use_amp once.",
+            )
 
-        check.true(len(self.models) == 0, "Please call wrap_scaler before wrap_model.")
+        if self.models:
+            raise det.errors.InvalidExperimentException(
+                "Please call wrap_scaler before wrap_model.",
+            )
 
-        check.true(
-            torch.cuda.is_available(),
-            "Mixed precision training (AMP) is supported only on GPU slots.",
-        )
+        if not torch.cuda.is_available():
+            raise det.errors.InvalidExperimentException(
+                "Mixed precision training (AMP) is supported only on GPU slots.",
+            )
```

Comment:
Again, mention that it's the call to `wrap_scaler` that triggered this.

Also, this check should only be triggered if the scaler is enabled, right?

Reply:
I suppose so. Should we make this method a no-op if `scaler.is_enabled()` is false?

Reply:
no I think the other checks are all structural, like the order in which you call things, and those are fine to enforce in all cases I think.

Reply:
So aside from mentioning `wrap_scaler` (done), leave things as is?

Reply:
I understand this now. Done.